### DISCUSSION TITLE
PerUserOrMachine will not do "Program Files" directory for all users

### DISCUSSION THREAD

**Discussion by Captnwalker1** [wixtoolset/issues#7289] [Questions]
Trying to implement a dual-scope package with WiX4, using the new `scope="PerUserOrMachine"`. However every time  I would select the PerMachine option it would still give me a `%LOCALAPPDATA%` path e.x.: `C:\Users\USERNAME\AppData\Local\Programs\PerUserOrMachine-Example`

See my example project with workaround. Just add scope to Package tag and it still breaks even with the workaround.
https://github.com/Captnwalker1/Wix4-PerUser-OR-PerMachine-Example

## Bug

* Which version of WiX are you building with?
4.0.0-rc.4+dea44a7d

* Which version of Visual Studio are you building with (if any)?
Microsoft Visual Studio Enterprise 2022 Version 17.5.0

* Which version of the WiX Toolset Visual Studio Extension are you building with (if any)?
HeatWave for VS2022 0.9.5

* Which version of .NET are you building with?
7.0.201

* If the problem occurs when installing your packages built with WiX, what is the version of Windows the package is running on?
Version	10.0.19044 Build 19044

* Describe the problem and the steps to reproduce it.
Add Scope="perUserOrPerMachine" to wix project, if you use [ProgramFilesFolder] it will then only try to go within %localappdata%.
See Example Project with workaround for InstallScopeDlg: https://github.com/Captnwalker1/Wix4-PerUser-OR-PerMachine-Example

* Describe the behavior you expected and how it differed from the actual behavior.
Expecting a path with %localappdata% when picking perUser install, but not when selecting perMachine install. Should be like C:\Program Files\ or C:\Program Files(x86)\

**Comment by barnson** [comment]
`Scope="perUserOrMachine"` works as expected from the command line:

```
msiexec /l*vx permachine.log /i DualPurposePackage.msi ALLUSERS=1

MSI (c) (CC:64) [15:00:21:909]: Dir (target): Key: INSTALLFOLDER	, Object: C:\Program Files (x86)\TODO Manufacturer DualPurposePackage\
```

-v-

```
msiexec /l*vx peruser.log /i DualPurposePackage.msi

MSI (c) (78:10) [15:02:35:842]: Dir (target): Key: INSTALLFOLDER	, Object: C:\Users\User Name\AppData\Local\Programs\TODO Manufacturer DualPurposePackage\
```

FWIW, WixUI_Advanced predates dual-purpose packages so doesn't support them.

**Reply by Captnwalker1** [reply]
So there is no way to use any UI with `Scope="PerUserOrMachine"`?

I know in enterprise its common to use msiexec but most end-users will just be running the msi directly.

**Reply by barnson** [reply]
So far, nobody's done the work to make WixUI_Advanced work with dual-purpose packages.

**Reply by robmen** [reply]
@Captnwalker1 if this is something you'd like to work on in WiX v5, I'd recommend opening an issue to start the [WIP process](https://wixtoolset.org/docs/development/wix-improvement-proposal/).

**Reply by gdams** [reply]
Hi there, I just wanted to check if this is still the case? The Adoptium project is migrating from with v3 to v5 and this feels like a pretty major regression. Are there any workarounds such as a permissions elevation dialogue to switch between perUser and perMachine?

**Reply by robmen** [reply]
It's not a regression. WixUI_Advanced was developed before Windows Installer introduced per-user/per-machine switching packages. The point is no one has made an effort to make it work <strike>better</strike> at all.

None of my customers use this feature of the Windows Installer (I am not sure any even use `WixUI_Advanced` either), so I don't personally prioritize it.
